packages/web/src/components/date/DateRange.js

```diff
@@ -61,7 +61,7 @@ class DateRange extends Component {
 	}
 
 	componentDidUpdate(prevProps) {
-		if (!isEqual(this.props.value, prevProps.value)) {
+		if (!isEqual(JSON.stringify(this.props.value), JSON.stringify(prevProps.value))) {
```

Comment:
@anik-ghosh-au7 I'm not sure if this change would fix the issue because `isEqual` does the same thing. It'll be best if you can share the kind of comparison that fails.
https://www.loom.com/share/0105a5a6b3a7490faaf696edfeb1f0fc

Reply:
@bietkul https://www.loom.com/share/84cceda1046243e982655727eed14d3c